src/lib/alarmMgr/AlarmManager.cpp

```diff
@@ -0,0 +1,240 @@
+*
+* AlarmManager::dbErrorReset - 
+*
+* Returns false if no action is taken/necessary, otherwise, true is returned.
+*/
+bool AlarmManager::dbErrorReset(void)
+{
+  if (dbOk == true)
+  {
+    return false;
+  }
+
+  ++dbErrors;
+  LM_E(("Releasing Database alarms"));
+  dbOk = true;
+  return true;
+}
+
+
+
+/* ****************************************************************************
+*
+* AlarmManager::notificationError - 
+*
+* Returns false if no action is taken/necessary, otherwise, true is returned.
+*
+* NOTE
+* The number of notificationErrors per url is maintained in the map.
+* Right now that info is not used, but might be in the future.
+* To do so, we'd need another counter as well, to not forget the accumulated 
+* number each time the notificationErrors are reset.
+*/
+bool AlarmManager::notificationError(const std::string& url, const std::string& details)
+{
+  std::map<std::string, int>::iterator iter = notificationV.find(url);
+
+  ++notificationErrors;
+
+  if (iter != notificationV.end())  // Already exists - add to the 'url-specific' counter
+  {
+    iter->second += 1;
+  }
+  else
+  {
+    notificationV[url] = 1;
+  }
+
+  if (notificationErrorLogInterval != 0)
+  {
+    if ((notificationErrors % notificationErrorLogInterval) == 1)
+    {
+      LM_W(("Notification Error [%d] for %s: %s", notificationErrors, url.c_str(), details.c_str()));
+    }
+  }
+
+  return true;
+}
+
+
+
+/* ****************************************************************************
+*
+* AlarmManager::notificationErrorReset - 
+*
+* Returns false if no action is taken/necessary, otherwise, true is returned.
+*/
+bool AlarmManager::notificationErrorReset(const std::string& url)
+{
+  if (notificationV.find(url) == notificationV.end())  // Doesn't exist
+    return false;
+  
+  notificationV.erase(url);
+  LM_W(("Releasing Notification Error alarm for %s", url.c_str()));
+
+  return true;
+}
+
+
+
+/* ****************************************************************************
+*
+* AlarmManager::badInput - 
+*
+* Returns false if no action is taken/necessary, otherwise, true is returned.
+*
+* NOTE
+* The number of badInputs per IP is maintained in the map.
+* Right now that info is not used, but might be in the future.
+* To do so, we'd need another counter as well, to not forget the accumulated 
+* number each time the badInputs are reset.
+*/
+bool AlarmManager::badInput(const std::string& ip, const std::string& details)
+{
+  std::map<std::string, int>::iterator iter = badInputV.find(ip);
+
+  ++badInputs;
+
+  if (iter != badInputV.end())  // Already exists - add to the 'ip-specific' counter
+  {
+    iter->second += 1;
+  }
+  else
+  {
+    badInputV[ip] = 1;
+  }
+
+  if (badInputLogInterval != 0)
+  {
+    if ((badInputs % badInputLogInterval) == 1)
+    {
+      LM_W(("Bad Input [%d] for %s: %s", badInputs, ip.c_str(), details.c_str()));
+    }
+  }
+
+  return true;
```

Comment:
You are so right - a few 'return false' are missing.
The Reset functions are ok, but the other three lack the 'return false'